views/partials/dashboardResults.ejs

```diff
@@ -1,103 +1,158 @@
+                    <%= dayjs(users_applications[i].createdDate).format('DD-MM-YYYY') %>
+                </td>
+                <td>
+                    <% if (users_applications[i].applicationtype === 'e-Apostille') { %>
+                        <a class="govuk-link"
+                            data-testid="eApp-ref-link"
+                            href="/open-eapp/<%= users_applications[i].unique_app_id %>">
+                            <%= users_applications[i].unique_app_id %>
+                        </a>
+                    <% } else { %>
+                        <a class="govuk-link" href="/open-application/<%= users_applications[i].unique_app_id %>">
+                            <%= users_applications[i].unique_app_id %>
+                        </a>
+                    <% } %>
+                </td>
+                <td class="hide-col-sm hide-col-md">
+                    <%= users_applications[i].user_ref || 'Not available' %>
+                </td>
+                <td>
+                    £<%= parseFloat(users_applications[i].payment_amount + 0).toFixed(2) %>
+                </td>
+                <td class="hide-col-sm hide-col-md">
+                    <%= users_applications[i].applicationtype %>
+                </td>
+                <td>
+                    <strong class="govuk-tag <%= users_applications[i].app_status.colorClass %>">
+                        <%= users_applications[i].app_status.text %>
+                    </strong>
+                    <div>
+                        <% if (users_applications[i].app_status.text==='Completed' && users_applications[i].rejected_docs> 0) {%>
+                            <%= users_applications[i].rejected_docs %>
+                            PDF<% if (users_applications[i].rejected_docs> 1) {%>s<%}%> rejected
+                        <% } %>
+                        <% if (users_applications[i].tracking_ref) { %>
+                            <% if (users_applications[i].main_postcode) { %>
+                                <a href="https://www.dxdelivery.com/?tn=<%=users_applications[i].tracking_ref%>&tpc=<%=users_applications[i].main_postcode%>" target="_blank">
+                                    Tracking ref: <br>
+                                    <%= users_applications[i].tracking_ref %>
+                                </a>
+                            <% } else { %>
+                                <a href="https://www.dxdelivery.com/consumer/my-tracking/" target="_blank">
+                                    Tracking ref: <br>
+                                    <%= users_applications[i].tracking_ref %>
+                                </a>
+                            <% } %>
+                        <% } %>
+                    </div>
+                </td>
+            </tr>
+        <% } %>
     </tbody>
 </table>
 
 
 
 <div class="pager">
-    <p class="pager-summary"><%= paginationMessage %></p>
+    <p class="pager-summary">
```

Comment:
Not sure if possible in this repo but you may want to consider creating a fragment for the pagination details so you can just insert a reference / import to it here.  It'll then make the dashboardResults page more specific to the result data rather than the boilerplate stuff like pagination functionality.

Reply:
yeah pretty much.  Essentially it's a way to extract out a block of logic / functionality to a separate file so you're left with a single "import" line in the parent which allows you to just focus on the business logic rather than perhaps boilerplate code like headers and footers.  I found this example of using partials for EJS template engine which it looks like this app is using: https://www.digitalocean.com/community/tutorials/how-to-use-ejs-to-template-your-node-application